### ISSUE TITLE
Different hover location in autopy after 3.0.1

### ISSUE THREAD

**Issue by pevogam** [autopilot-rs/autopy#52]
Some of our unit tests using autopy backend fail starting from version 3.0.1 and all the way until version 4.0.0. In particular, we test mouse hover locations which seem to be different from versions prior to and including 3.0.0 and any recent versions. Did anything significant about mouse behavior change in this version transition (from 3.0.0 to 3.0.1)?

**Comment by msanders** [comment]
3.0.1 fixes a bug where `mouse.location` and `mouse.move` were inconsistent on Windows and X11. See https://github.com/autopilot-rs/autopy/issues/42.

Previously there were tests missing for this so the bug was missed but they have since [been added](https://github.com/autopilot-rs/autopilot-rs/blob/1bd43f08561755ae8420ad33f4d378f8c0602bd7/src/mouse.rs#L387).

**Comment by pevogam** [comment]
Interesting! I thought it should be something like this. The strange thing on my side though is that the failing unit test that alerted me complains about large difference between the location hovered by autopy and the location detected by OpenCV:

```
======================================================================

FAIL: test_hover (test_region_control.RegionTest)

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/usr/lib/python3.7/site-packages/guibot/tests/test_region_control.py", line 130, in test_hover

    self.assertAlmostEqual(match.target.x, self.region.mouse_location.x, delta=1)

AssertionError: 548 != 506 within 1 delta (42 difference)
```

with code as

```python
    def test_hover(self):
        self.show_application()

        match = self.region.find('shape_green_box')
        self.region.hover(match.target)
        self.assertAlmostEqual(match.target.x, self.region.mouse_location.x, delta=1)
        self.assertAlmostEqual(match.target.y, self.region.mouse_location.y, delta=1)

        # hover over coordinates in a subregion
        match = match.find('shape_green_box')
        self.assertAlmostEqual(match.target.x, self.region.mouse_location.x, delta=1)
        self.assertAlmostEqual(match.target.y, self.region.mouse_location.y, delta=1)

        self.close_windows()
```

This means that either the OpenCV should also perceive the screen wrong on the X11 system the test is ran on or the consistency with Windows leaves some significant difference for other screenshot taking software on Linux. I could look at some screenshots on where autopy hovers, this excerpt was from a CI.

**Comment by pevogam** [comment]
Hmmm but then I obtain this only on some screens so I guess it is something I will have to investigate further. Thanks for all your feedback!

**Comment by msanders** [comment]
Thanks for reporting! Let me know what you find out.

**Comment by pevogam** [comment]
Turns out I didn't reproduce this issue on other screens exactly because the `scale` attribute is `1.0` there and the differences in your recent fix were not detectable. Now that I got it indeed, your fix for the mouse location makes sense. I used to always divide by `scale` in the past (including when obtaining the mouse location) since all I wanted was for the unit tests (based on pixels instead of dots) to work with the backend. Now that it is fixed I had to do the reverse conversion for the mouse location and everything works as expected.

Thanks for pointing at the relevant commit and original issue.
